pkg/controller/deployment/sync.go

```diff
@@ -454,7 +454,7 @@ func (dc *DeploymentController) cleanupDeployment(ctx context.Context, oldRSs []
 	for i := int32(0); i < diff; i++ {
 		rs := cleanableRSes[i]
 		// Avoid delete replica set with non-zero replica counts
-		if rs.Status.Replicas != 0 || *(rs.Spec.Replicas) != 0 || rs.Generation > rs.Status.ObservedGeneration || rs.DeletionTimestamp != nil {
+		if rs.Status.Replicas != 0 || *(rs.Spec.Replicas) != 0 || rs.Generation > rs.Status.ObservedGeneration {
```

Comment:
Why do you want to remove a condition that ensure we ignore replicasets which are already marked for deletion?